invisible_cities/cities/detsim_test.py

```diff
@@ -0,0 +1,103 @@
+import os
+import shutil
+import numpy  as np
+import pandas as pd
+import tables as tb
+
+from .. cities.detsim  import detsim
+from .. core.configure import configure
+
+from invisible_cities.core  import system_of_units as units
+
+
+def test_detsim_contains_all_tables(ICDATADIR, config_tmpdir):
+
+    PATH_IN  = os.path.join(ICDATADIR    , "nexus_new_kr83m_fast.newformat.sim.h5")
+    PATH_OUT = os.path.join(config_tmpdir, "detsim_test.h5")
+    conf = configure('detsim $ICTDIR/invisible_cities/config/detsim.conf'.split())
+
+    buffer_length     = 800 * units.mus
```

Comment:
You seem to be reusing `conf['buffer_length']` rather than changing all its values, so you can either update it directly or create a local reference to it and update it since those are all mutable objects

Reply:
> create a local reference to it and update it since those are all mutable objects

Didactic point ... I think that 'local reference' is an unfortunate choice of words: you should have said 'local copy'. 'Local reference' is not strictly incorrect, but it's ambiguous and fails to convey the important point. I'd go as far as saying that its most obvious interpretation is the exact opposite of what you mean!

It would be good if you all developed a habit of getting nervous when something is mutated, especially something that you got from somewhere else. Mutable shared state is the root of all evil in programming. Sometimes we need to mutate for pragmatic reasons, but do avoid mutation wherever pragmatically possible. In this case, we're talking about copying a smallish dictionary in a test that does IO (hint: IO is veeeeeeery slow compared to in-memory operations).

If you make a local copy of this dict, you can mutate it to your heart's content, and nobody ever has to worry that you're breaking things in some other part of the code which holds another reference to the thing you are mutating.